posts/integrating-civic-into-a-static-serverless-website-part-1-of-2.md

```diff
@@ -0,0 +1,231 @@
+
+   For this project, I'm using [https://color.tylerjohnhaden.com](https://color.tylerjohnhaden.com) for the website and [https://api.color.tylerjohnhaden.com](https://api.color.tylerjohnhaden.com) for the API endpoint. If this app is still up when you are reading this, you can see the demo by going to first link.
+
+### Create an S3 bucket and upload initial static files
+
+Since CloudFront takes the longest to spin up, let us speed up the process by skipping any integration yet. We can put at least an `index.html` in a public S3 bucket. Here is an [intro to getting started with S3](https://docs.aws.amazon.com/quickstarts/latest/s3backup/welcome.html). We will not need to configure static website hosting because CloudFront will take care of that (the S3 built-in hosting does not support SSL for custom domains).
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8" />
+    <title>Color Identity!</title>
+    <link rel="icon" href="color.png" />
+</head>
+<body onload="requestAnonymousIdentity()">
+</body>
+</html>
+```
+
+We will define `requestAnonymousIdentity()` later on. And what's a website without a [fun favicon](https://opensea.io/assets/0xaefa27a665d48e19c38437ba7135c8107bb5928f/17)?
+
+### Create a CloudFront distribution with your domain
+
+Using the S3 bucket as an origin, spin up a new distribution. You will need to use the certificate previously created and add the domain you want the site to use as a CNAME in CloudFront.
+
+You will also want to make sure that CloudFront has access to the bucket. This is done through "Origin Access Identity" and you can have CloudFront create a new identity and register it with the S3 origin bucket. Otherwise, you will get an `Access Denied` response when hitting the domain.
+   
+Finally, to get that domain to route to your distribution, we can add an alias to Route53. The easiest way to do this is to copy the domain name of the CloudFront distribution (not its CNAME), and setting it as the Alias Target in Route53. It should look something like this:
+
+![missing pic: Civic Alias](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/02/civic_alias.png)
+
+p.s. This does not need to be a sub-domain. I just happen to be using the root domain `tylerjohnhaden.com` for something else.
+
+Once created, it will take a while to spin up, so let us continue with the rest of the stack. Uploading a new `index.html` or other resource to S3 takes a few seconds, so we can easily update our client side code later.
+
+# Setup your Civic integration
+
+In order to use the developer tools for Civic, we need to use our identity to create an account and Civic app. [Login](https://integrate.civic.com/login) to the their developer's portal. This will also give you an identity that can be used to test out the app later on. They have straight forward instructions on downloading the mobile app and verifying your identity.
```

Comment:
```suggestion
In order to use the developer tools for Civic, we need to use our identity to create an account and Civic app. [Login](https://integrate.civic.com/login) to their developer's portal. This will also give you an identity that can be used to test out the app later on. They have straightforward instructions on downloading the mobile app and verifying your identity.
```